x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_header.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { useValues } from 'kea';
+
+import queryString from 'query-string';
+import moment from 'moment';
+
+import { i18n } from '@kbn/i18n';
+import {
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelect,
+  EuiDatePickerRange,
+  EuiDatePicker,
+  EuiButton,
+} from '@elastic/eui';
+
+import { KibanaLogic } from '../../../../shared/kibana';
+import { LogRetentionTooltip, LogRetentionOptions } from '../../log_retention';
+
+import { AnalyticsLogic } from '../';
+import { SERVER_DATE_FORMAT } from '../constants';
+import { convertTagsToSelectOptions } from '../utils';
+
+interface Props {
+  title: string;
+}
+export const AnalyticsHeader: React.FC<Props> = ({ title }) => {
+  const { allTags } = useValues(AnalyticsLogic);
+  const { history } = useValues(KibanaLogic);
+
+  // Parse out existing filters from URL query string
+  const { start, end, tag } = queryString.parse(history.location.search);
+  const [startDate, setStartDate] = useState(
+    start ? moment(start, SERVER_DATE_FORMAT) : moment().subtract(6, 'days')
+  );
+  const [endDate, setEndDate] = useState(end ? moment(end, SERVER_DATE_FORMAT) : moment());
+  const [currentTag, setCurrentTag] = useState((tag as string) || '');
+
+  // Set the current URL query string on filter
```

Comment:
Non-blocking

> // Set the current URL query string on filter

General code style comment. I think you can usually drop comments like this that say "what" the code is doing, and favor a well named method if it's not clear as is.

Reply:
I think I might end up redoing this section anyway based on your below comment, but was interested in generally rambling about the general concept of comments!

I def know it's a subjective topic (on one end: the Java folks who would probably name the below method `setUrlSearchQueryStringOnApplyFilters` :trollface:). I think in this scenario my thinking was that there's two main things this component does - pull state from the query params and push state to the query params - and I wanted two blocks of code that reflected that + comment headings clarifying that as well.

I will say as a dev I probably do lean more towards over-commenting than under-commenting personally. It doesn't cost memory/performance, it potentially helps other developers understand my thinking/organization etc., and if it's really unnecessary someone will come along and remove it eventually. However, if I ever do start tilting too much to the other end of the scale and doing things like

```
// Do something
doSomething();
```

please definitely start flagging that to me lol.